db/src/main/java/net/bible/android/database/WorkspaceEntities.kt

```diff
@@ -40,7 +41,7 @@ val json = Json {
 class WorkspaceEntities {
     data class Page(
         val document: String?,
-        val key: String?,
+        val key: Key?,
```

Comment:
Looks like you might need to migration for this yet? I get crash when trying to open app:
```
2020-11-28 11:02:02.643 6858-6858/net.bible.android.activity I/MainBibleActivity: Creating MainBibleActivity
2020-11-28 11:02:02.649 6858-6858/net.bible.android.activity I/net.bible.android.view.activity.page.MainBibleActivity: onCreate:net.bible.android.view.activity.page.MainBibleActivity@131fae6
2020-11-28 11:02:02.692 6858-6858/net.bible.android.activity D/SpeakSettings: SpeakSettings loaded! SpeakSettings(synchronize=true, replaceDivineName=false, autoBookmark=false, restoreSettingsFromBookmarks=false, playbackSettings=PlaybackSettings(speakChapterChanges=true, speakTitles=true, speakFootnotes=false, speed=100, bookId=null, bookmarkWasCreated=null, verseRange=null), sleepTimer=0, lastSleepTimer=10, queue=true, repeat=false, numPagesToSpeakId=0)
2020-11-28 11:02:02.776 6858-6858/net.bible.android.activity D/net.bible.android.control.navigation.DocumentBibleBooksFactory: Initialising DocumentBibleBooksFactory cache
2020-11-28 11:02:02.808 6858-6858/net.bible.android.activity D/WinRepository: onLoadDb for workspaceId=1
2020-11-28 11:02:02.809 6858-6858/net.bible.android.activity D/WinRepository: clear false
2020-11-28 11:02:02.811 6858-6858/net.bible.android.activity D/AndroidRuntime: Shutting down VM
2020-11-28 11:02:02.812 6858-6858/net.bible.android.activity E/AndroidRuntime: FATAL EXCEPTION: main
    Process: net.bible.android.activity, PID: 6858
    java.lang.RuntimeException: Unable to start activity ComponentInfo{net.bible.android.activity/net.bible.android.view.activity.page.MainBibleActivity}: java.lang.IllegalArgumentException: bad base-64
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3449)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601)
        at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:85)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
     Caused by: java.lang.IllegalArgumentException: bad base-64
        at android.util.Base64.decode(Base64.java:163)
        at android.util.Base64.decode(Base64.java:138)
        at android.util.Base64.decode(Base64.java:120)
        at net.bible.android.database.Converters.strToKey(AppDatabase.kt:102)
        at net.bible.android.database.WorkspaceDao_Impl.pageManager(WorkspaceDao_Impl.java:2508)
        at net.bible.android.control.page.window.WindowRepository.loadFromDb(WindowRepository.kt:377)
        at net.bible.android.control.page.window.WindowRepository.initialize(WindowRepository.kt:110)
        at net.bible.android.view.activity.page.MainBibleActivity.onCreate(MainBibleActivity.kt:222)
        at android.app.Activity.performCreate(Activity.java:8000)
        at android.app.Activity.performCreate(Activity.java:7984)
```

Reply:
yes... I'm still not sure if I leave it like this. base64 binary serialization is not so desirable option.
Though this affects only dictionaries and gen books.

Reply:
I was considering also to use Gson. But it's bit more difficult.